### PR TITLE
React capstone - Planètes-et-astéroïdes

### PR DESCRIPTION
**Pull request by Sboursen**
## Description

> **Planètes-et-astéroïdes** is a web app powered by the [The solar system API](https://api.le-systeme-solaire.net/en/) that displays metrics relative to the planets, satellites, and asteroids that constitute our solar system. In addition to the ability to navigate between the different celestial bodies, it also offers a powerful _search by name_ functionality. It is made as a capstone project of the React&Redux module.

### General requirements

- [x] There are no linter errors.
- [x] Followed the **Github flow**.
- [x] The documentation is professional.

### HTML/CSS and JavaScript requirements

- [x] Followed the list of [best practices for HTML & CSS](https://github.com/microverseinc/curriculum-html-css/blob/main/articles/html_css_best_practices.md).
- [x] Followed the list of [best practices for JavaScript](https://github.com/microverseinc/curriculum-html-css/blob/main/articles/javascript_best_practices.md)

### Built With

- Tailwindcss
- React
- Redux
- Redux-thunk
- React-router
- Redux-logger (tooling)
- React testing library (unit testing)
- The solar system API

### Milestones

- [x] Milestone 1: **React app setup with redux using the duck pattern**
- [x] Milestone 2: **Set up the app navigation using react-router**
- [x] Milestone 3: **Create an API layer using the Solar System API**
- [x] Milestone 4: **Create the Redux store including the action creators and the reducers**
- [x] Milestone 5: **Integrate the redux store into the app and display the data in the Home route**
- [x] Milestone 6: **Implement Details route functionalities**
- [x] Milestone 7: **Implement Search route and functionality**
- [x] Milestone 8: **Create the About route**
- [x] Milestone 9: **Finalize the styling**
- [x] Milestone 10: **Implement additional tests**

### Additional notes

> I apologize for the submission delays. It was due to some difficulties handling the **API** and the Tests.

> We made a [App demo](https://www.loom.com/share/150f559ebeb74d67a600ecd27e333c36) for my website.

> We easy access to our website please use this [Live link](https://planetes-et-asteroides.netlify.app/)

**_Thank you for taking the time to review this pull request._**

### CODE REVIEW

File: src/routes/details/Details.js [hunk 1]
```diff
@@ -0,0 +1,9 @@
+import { Outlet } from 'react-router-dom';
+
+export default function Details() {
+  return (
+    <section className="mt-6 rounded w-full">
+      <Outlet />
+    </section>
+  );
```
Comment (micheaol) [inline src/routes/details/Details.js:8]: Please ensure that there's a link from the Detail page to the `Home` page by clicking the "Back" button as in the original design.
Your Implementation | Design
:---:|:---:
image | image

File: src/routes/home/Home.js [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+    return (
+      <section className="mt-12 px-1 rounded w-full">
+        <div className="w-5/6 h-48 flex flex-col justify-around items-center shadow-sm rounded p-2 shadow-gray-600 mx-auto">
+          <h1 className="text-7xl text-center">Loading the universe</h1>
+          <span className="animate-spin flex flex-row justify-center text-7xl">
+            <BiLoaderAlt />
+          </span>
+        </div>
+
+        <ul className="grid xs:grid-cols-2 md:grid-cols-3 lg:grid-cols-4 items-center justify-items-center p-3 gap-3">
```
Comment (micheaol) [inline src/routes/home/Home.js:26]: Please ensure that you follow the [design guidelines](https://www.behance.net/gallery/31579789/Ballhead-App-(Free-PSDs)). 

##Home
Your Implementation | Design
:---:|:---:
image | image


## Detail page
Your Implementation | Design
:---:|:---:
image | image

**Review by micheaol — CHANGES_REQUESTED** [review]
Hi @Sboursen , 

Good job so far!
There are some issues that you still need to work on to prepare your project for the final evaluation but you are almost there!

## TO HIGHLIGHT: 
- Your project is professional  💯 
- Your repo is professional 💯
- Well structured files 💯
- You set up linter files and no errors. 💯
-  Great job writing the tests 💯
- Great presentation. 💯


## Suggested changes
_Check the comments under the review._

You can use as many of my suggestions as you want. If there is anything you would like to skip - feel free to do that. However, I strongly recommend you to take them into account as they can make your code better._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by Sboursen** [comment]
Hi @micheaol,

Thank you for the feedback.
I want to explain that you can go back to **Home** from any page by clicking on the Home link!

So Are you suggesting that I just put an icon instead of the text Home?

Best,

**Comment by Sboursen** [comment]
I also added a search feature
Are you suggesting I remove it?

**Comment by Sboursen** [comment]
Hi 
I made the requested changes
here is the new interface for the mobile version:

image

**Review by emmanuellekamwa — APPROVED** [review]
**STATUS: APPROVED 🍾🎉✔**
Hello @Sboursen 👋
Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Highlights

- You asked for the changes in a polite and friendly way
- The use of Github inline comments feature is used properly.
- You documented your work professionally. ✔

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by Sboursen** [comment]
Hi @emmanuellekamwa 
Thank you for the approval!
Best

**Pull request merged**